pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -291,8 +302,16 @@ def _del_model(self, filepath: str):
             self._fs.rm(filepath)
             log.debug(f"Removed checkpoint: {filepath}")
 
-    def _save_model(self, filepath: str, trainer, pl_module):
-        # Todo: required argument `pl_module` is not used
+    def _save_model(self, trainer, filepath: str):
+        if trainer.training_type_plugin.rpc_enabled:
+            # RPCPlugin manages saving all model states
```

Comment:
The RPC Plugin could override save_function from the trainer as TPUSpawn does.

Reply:
This was my point with the `TODO` right below. But didn't see how to easily do it since `rpc_save_model` is more of a wrapper than a `save_function` implementation.

If you want to take this refactor, go for it. It doesn't have to be in this PR as we also have this issue in master